lib/internal/observable.js

```diff
@@ -0,0 +1,898 @@
+              if (queue.length > 0) {
+                return processFlatMap(queue.shift());
+              }
+              activeInnerSubscription = false;
+              if (outerSubscriptionHasCompleted) subscriber.complete();
+            },
+          }, {
+            signal: subscriber.signal,
+          });
+        } catch (err) {
+          subscriber.error(err);
+        }
+      };
+
+      this.subscribe({
+        next(value) {
+          if (activeInnerSubscription) {
+            queue.push(value);
+            return;
+          }
+          activeInnerSubscription = true;
+          processFlatMap(value);
+        },
+        error(error) {
+          subscriber.error(error);
+        },
+        complete() {
+          outerSubscriptionHasCompleted = true;
+          if (!activeInnerSubscription && queue.length === 0) {
+            subscriber.complete();
+          }
+        },
+      }, {
+        signal: subscriber.signal,
+      });
+    });
+  }
+
+  /**
+   * @param {Mapper} mapper
+   * @returns {Observable}
+   */
+  switchMap(mapper) {
+    return new Observable((subscriber) => {
+      let idx = 0;
+      let outerSubscriptionHasCompleted = false;
+      let activeInnerAbortController = null;
+
+      const processSwitchMap = (value) => {
+        try {
+          let mappedResult = mapper(value, idx++);
+          let innerObservable = Observable.from(mappedResult);
+          innerObservable.subscribe({
+            next(value) { subscriber.next(value); },
+            error(error) { subscriber.error(error); },
+            complete() {
+              if (outerSubscriptionHasCompleted) {
+                return subscriber.complete();
+              }
+              activeInnerAbortController = null;
+            }
+          }, {
+            signal: AbortSignal.any([
+              activeInnerAbortController.signal,
+              subscriber.signal,
+            ]),
+          });
+        } catch (err) {
+          subscriber.error(err);
+        }
+      };
+
+      subscriber.subscribe({
+        next(value) {
+          if (activeInnerAbortController !== null) {
+            activeInnerAbortController.abort();
+          }
+          activeInnerAbortController = new AbortController();
+          processSwitchMap(value);
+        },
+        error(error) { subscriber.error(error); },
+        complete() {
+          outerSubscriptionHasCompleted = true;
+          if (activeInnerAbortController == null) subscriber.complete();
+        },
+      }, {
+        signal: subscriber.signal,
+      });
+    });
+  }
+
+  /**
+   * @param {VoidCallback} callback
+   * @returns {Observable}
+   */
+  finally(callback) {
+    throw new ERR_METHOD_NOT_IMPLEMENTED('Observable.prototype.finally');
+  }
+
+  /**
+   * @param {SubcribeOptions} SubscribeOptions
+   * @param {*} options
+   * @returns {Promise<any[]>}
+   */
+  toArray(options = {}) {
+    const { promise, resolve, reject } = createDeferredPromise();
```

Comment:
```suggestion
    const { promise, resolve, reject } = Promise.withResolvers();
```
nitpic

Reply:
`Promise.withResolvers` is not available in the primordial due to not yet graduated from harmony features.